public/app/features/alerting/unified/rule-list/hooks/usePaginatedPrometheusGroups.tsx

```diff
@@ -73,5 +73,14 @@ export function usePaginatedPrometheusGroups<TGroup extends PromRuleGroupDTO>(
     return groups.slice((currentPage - 1) * pageSize, currentPage * pageSize);
   }, [groups, currentPage, pageSize]);
 
-  return { isLoading: fetchInProgress, page: groupsPage, nextPage, previousPage, canMoveForward, canMoveBackward };
+  return {
+    isLoading: fetchInProgress,
+    groups,
```

Comment:
returning the entire `groups` here so we can keep grabbing more data instead of dividing it into pages.